### DISCUSSION TITLE
JavaScript Support

### DISCUSSION THREAD

**Discussion by thomasdarimont** [keycloak/keycloak#10190] [Archive]
The Nashorn JS engine has eventually been removed in  Java 15. This makes the Javascript support not usable with a plain JDK > 14.
However, it is possible to add the [self-contained nashorn-core library](https://search.maven.org/artifact/org.openjdk.nashorn/nashorn-core/15.3/jar) manually to the `$KC_HOME/providers` folder. 
After a config rebuild (via `kc.sh build` or `--auto-build`) one can use the JavaScript support with JDK versions > JDK 14.
I just verified this with the latest JDK 17.

The scripting feature can be enabled with `--features=scripts` if one wants to be able to edit scripts in the admin ui then enable the `upload-scripts` feature as well, via `--features=scripts,upload-scripts`.

I think it would be a good idea to add the JS engine to the distribution by default - once the JDK will be updated to JDK17+, WDYT?

**Comment by stianst** [comment]
See https://github.com/keycloak/keycloak/issues/9945

**Comment by stianst** [comment]
Moving this to archive as we already have an issue open to address Nashorn removal related to supporting OpenJDK 17.

**Comment by Arulaln-AR** [comment]
@thomasdarimont 

Need help in enabling the script feature. I have mapped the script feature using the environment variable. Attached is the complete dockerfile. Built an image based on that and still under Authentication - > Add Execution step, I don't see an script option to create Javascript.

Do I need to add the javascript as well under the /opt/keycloak/providers/ during the build stage itself.
[Dockerfile.txt](https://github.com/keycloak/keycloak/files/9234910/Dockerfile.txt)
